asynciterator.ts

```diff
@@ -491,7 +491,7 @@ export class AsyncIterator<T> extends EventEmitter {
     After this operation, only read the returned copies instead of the original iterator.
     @returns {module:asynciterator.AsyncIterator} A new iterator that contains all future items of this iterator
   */
-  clone(): AsyncIterator<T> {
+  clone(): ClonedIterator<T> {
```

Comment:
Is there a reason we want to give the more specific class? I thought to keep `ClonedIterator` (more or less) internal.

Reply:
I just checked again, and it was not really needed.

But I would still consider keeping the more specific type though, as ClonedIterator (and TransformIterator) will have some more typed elements, such as `.source`.
But no strong opinion.

Reply:
👍 will keep